native/core/src/parquet/mod.rs

```diff
@@ -539,24 +537,16 @@ pub extern "system" fn Java_org_apache_comet_parquet_Native_currentBatch(
     e: JNIEnv,
     _jclass: JClass,
     handle: jlong,
-) -> jlongArray {
-    try_unwrap_or_throw(&e, |env| {
+    array_addr: jlong,
+    schema_addr: jlong,
+) {
+    try_unwrap_or_throw(&e, |_env| {
         let ctx = get_context(handle)?;
         let reader = &mut ctx.column_reader;
         let data = reader.current_batch();
-        let (array, schema) = data.to_spark()?;
+        data.move_to_spark(array_addr, schema_addr)?;
```

Comment:
nit: maybe we can just return `data.move_to_spark(array_addr, schema_addr)` so that we do not need `ok(())`?

Reply:
Okay.